lib/internal/buffer.js

```diff
@@ -1053,6 +1054,18 @@ function markAsUntransferable(obj) {
   obj[untransferable_object_private_symbol] = true;
 }
 
+// This simply checks if the object is marked as untransferable and doesn't
+// check the ability to be transferred.
+function isMarkedAsUntransferable(obj) {
+  if (obj == null)
+    return false;
+  // v8::Object::HasPrivate checks own properties only.
+  if (!ObjectPrototypeHasOwnProperty(obj, untransferable_object_private_symbol)) {
+    return false;
+  }
+  return obj[untransferable_object_private_symbol] === true;
```

Comment:
V8 doesn't go through the prototype for private symbols. This can just be:
```suggestion
  return obj[untransferable_object_private_symbol] !== undefined;
```